### PR TITLE
feat(dashboard): Add thumbnails to dashboard edit draggable chart list

### PR DESCRIPTION
**Pull request by codyml**
### SUMMARY

This PR adds thumbnails to the sidebar of draggable charts in the edit dashboard view and modifies the design of the cards with and without thumbnails enabled.  Thumbnails are shown when the `THUMBNAILS` feature flag is enabled.  If no thumbnail has been generated, the same placeholder that's used on the chart list card view is shown.  Other changes include:
- Layout updates for the text and the `Added` badge, when thumbnails are enabled and when they aren't.
- Text is truncated and shows a tooltip when it's too long to fit in the layout.
- The human-readable chart viz type is displayed instead of the machine-readable name.

### BEFORE/AFTER SCREENSHOTS OR ANIMATED GIF

**Before:**

https://user-images.githubusercontent.com/13007381/178516532-8830f90c-24fd-441c-a864-47c74b618df8.mov

**After (thumbnails enabled):**

https://user-images.githubusercontent.com/13007381/178517080-023dd336-7c35-4784-9efb-28c822a1bf08.mov

**After (thumbnails disabled):**

https://user-images.githubusercontent.com/13007381/178517519-89811718-34e0-4a97-85f3-17f040885f25.mov

### TESTING INSTRUCTIONS

**Note:** Thumbnail generation is currently broken (see #20673).  In addition, there's a bug preventing the chart metadata from being downloaded (see #20684), so even if thumbnails are generated, any items in the draggable chart list with missing metadata won't show thumbnails either.

1. Run Superset with feature flag `THUMBNAILS=False` and ensure that the updated cards look and work as expected when added and not added to the dashboard. 
2. Run Superset with `THUMBNAILS=True` and ensure that thumbnails appear (or at least that their placeholders do) as expected and that the updated cards look and work as expected when added and not added to the dashboard.

### ADDITIONAL INFORMATION

- [ ] Has associated issue:
- [x] Required feature flags: `THUMBNAILS`
- [x] Changes UI
- [ ] Includes DB Migration (follow approval process in [SIP-59](https://github.com/apache/superset/issues/13351))
  - [ ] Migration is atomic, supports rollback & is backwards-compatible
  - [ ] Confirm DB migration upgrade and downgrade tested
  - [ ] Runtime estimates and downtime expectations provided
- [x] Introduces new feature or API
- [ ] Removes existing feature or API

### CODE REVIEW

**Comment by codyml** [comment]
@kasiazjc here's the PR!

File: superset-frontend/src/dashboard/components/AddSliceCard/index.jsx [hunk 1]
```diff
@@ -111,14 +132,31 @@ function AddSliceCard({
...
   sliceName,
   style,
+  thumbnailUrl,
   visType,
 }) {
+  const showThumbnails = useMemo(
+    () => isFeatureEnabled(FeatureFlag.THUMBNAILS),
+    [],
+  );
+
```
Comment (zhaoyongjie) [inline superset-frontend/src/dashboard/components/AddSliceCard/index.jsx:142]: the value of `isFeatureEnabled(FeatureFlag.THUMBNAILS)` will not change when the Superset web application started, so the `useMemo` might be useless.

**Comment by zhaoyongjie** [comment]
@codyml please rebase master to fix the code coverage issue.

**Review by zhaoyongjie — APPROVED** [review]
looks great!

**Comment by rusackas** [comment]
I think we need to make some layout adjustments here... I'll ping you on a separate thread to save everyone the notification headaches :)

**Comment by zhaoyongjie** [comment]
I can't get chart snapshot on Dashboard edit list, but gets it on Charts List.

https://user-images.githubusercontent.com/2016594/176653301-6c6ccd50-06f4-47aa-bcec-b5de59d7b3eb.mov

**Comment by codyml** [comment]
> I can't get chart snapshot on Dashboard edit list, but gets it on Charts List.
> 
>  chart.snapshot.in.Dashboard.mov

Ah thanks, will look into this.

**Comment by codyml** [comment]
> I can't get chart snapshot on Dashboard edit list, but gets it on Charts List.
> 
>  chart.snapshot.in.Dashboard.mov

@zhaoyongjie Finally got around to looking into this, and it's another symptom of the missing metadata bug, which I'm going to tackle separately.  TBD if this is going to be the final fix, but if you comment out [this line](https://github.com/apache/superset/blob/8a5f3aca2a01c5da277556de5002f2d12c39dcff/superset-frontend/src/dashboard/actions/sliceEntities.js#L80) you should see all thumbnails and metadata appear.

**Comment by zhaoyongjie** [comment]
this PR should wait for merging [this](https://github.com/apache/superset/pull/20673).

**Review by geido — COMMENTED** [review]
Code looks good to me. Now waiting for a test env for manual testing. You might need to rebase to fix that codecov issue.

**Comment by zhaoyongjie** [comment]
@codyml the editor bar on the Dashboard cannot preview the thumbnails, but I can preview in the dashboard list.

image

**Comment by codyml** [comment]
> @codyml the editor bar on the Dashboard cannot preview the thumbnails, but I can preview in the dashboard list.
> 
> image

@zhaoyongjie Replied about this issue above, I think this is unrelated and fixed by https://github.com/apache/superset/pull/20684, which just got merged yesterday.  Just rebased over it, can or @geido run `/testenv up FEATURE_THUMBNAILS=true` and see if it's fixed?

File: superset-frontend/src/dashboard/components/AddSliceCard/index.jsx [hunk 2]
```diff
@@ -0,0 +1,276 @@
+/**
```
Comment (zhaoyongjie) [inline superset-frontend/src/dashboard/components/AddSliceCard/index.jsx:1]: Just noticed this is a JSX, we should migrate to TSX for the new component.

**Review by zhaoyongjie — APPROVED** [review]
awesome! I have got the screenshot in the edit bar on the dashboard. leave a no-blocking suggestion, migrate to TSX in separated PR or this PR.

image

**Comment by codyml** [comment]
@geido found an issue with metadata still being missing, need to look into it:
Screen Shot 2022-07-22 at 10 44 59 AM

**Comment by zhaoyongjie** [comment]
/testenv up FEATURE_THUMBNAILS=true FEATURE_THUMBNAILS_SQLA_LISTENERS=true

**Comment by zhaoyongjie** [comment]
/testenv up FEATURE_THUMBNAILS=true FEATURE_THUMBNAILS_SQLA_LISTENERS=true

**Comment by codyml** [comment]
@geido @zhaoyongjie Confirmed that missing metadata in ephemeral environment was due to ephemeral environment being spun up before PR was rebased over https://github.com/apache/superset/pull/20684.

**Review by rusackas — APPROVED** [review]
LGTM

**Pull request merged**